crates/modelardb_client/src/main.rs

```diff
@@ -62,72 +61,58 @@ const TRANSPORT_ERROR: &str = "transport error: no messages received.";
+        }
+        _ => {
+            // The errors are consciously ignored as the client is terminating.
+            let binary_path = env::current_exe().unwrap();
+            let binary_name = binary_path.file_name().unwrap().to_str().unwrap();
+
+            // Punctuation at the end does not seem to be common in the usage message of Unix tools.
+            eprintln!("Usage: {binary_name} [host or host:port] [query_file]",);
+            process::exit(1);
+        }
+    };
 
     // Execute the queries.
+    let flight_service_client = connect(host, port).await?;
     if let Some(query_file) = maybe_query_file {
-        execute_queries_from_a_file(flight_service_client, &query_file, &local_file_system).await
+        execute_queries_from_a_file(flight_service_client, &query_file).await
     } else {
         execute_queries_from_a_repl(flight_service_client).await
     }
 }
 
-/// Parse the command line arguments in `args` and return a triple with the host of the server to
-/// connect to, the port to connect to, and the file containing the queries to execute on the
-/// server. If one of these command line arguments is not provided it is replaced with [`None`].
-async fn parse_command_line_arguments(
-    mut args: Args,
-    local_file_system: &LocalFileSystem,
-) -> Result<(Option<String>, Option<u16>, Option<ObjectMeta>)> {
-    // Drop the path of the executable.
-    args.next();
-
-    // Parse command line arguments.
-    let mut maybe_host = None;
-    let mut maybe_port = None;
-    let mut maybe_query_file = None;
-
-    for arg in args {
-        let arg_path = Path::from(arg.as_str());
-        if let Ok(query_file) = local_file_system.head(&arg_path).await {
-            // Assumes all files contains queries.
-            maybe_query_file = Some(query_file);
-        } else if arg.contains(':') {
-            // Assumes anything with : is host:port.
-            let host_and_port = arg.splitn(2, ':').collect::<Vec<&str>>();
-            maybe_host = Some(host_and_port[0].to_owned());
-            maybe_port = Some(host_and_port[1].parse().map_err(|_| {
-                ModelarDbClientError::InvalidArgument(
-                    "Port must be between 1 and 65535.".to_owned(),
-                )
-            })?);
-        } else {
-            // Assumes anything else is a host.
-            maybe_host = Some(arg);
-        }
-    }
-
-    Ok((maybe_host, maybe_port, maybe_query_file))
+/// Parse the host and port in `maybe_host_port` and return a pair with the host of the server to
+/// connect to and the port to connect to. If the host is not included `DEFAULT_HOST` is used and if
+/// port is not included `DEFAULT_PORT` is used. Returns [`ModelarDbClientError`] if the port is not
+/// valid.
+fn parse_host_port(maybe_host_port: &str) -> Result<(&str, u16)> {
+    let mut parts = maybe_host_port.splitn(2, ':');
+    let host = parts.next().unwrap_or(DEFAULT_HOST);
+    let port = match parts.next() {
```

Comment:
Maybe use `Option::map_or()` instead of `match` here.

Reply:
Fixed.